packages/fx-core/src/component/code/tab/tabCode.ts

```diff
@@ -93,7 +93,7 @@ export class TabCodeProvider {
       fileNameReplaceFn: (name: string, data: Buffer): string =>
         removeTemplateExtReplaceFn(name, data).replace(
           TemplatePlaceHolders.ProjectFile,
-          ctx.projectSetting.appName
+          ctx.projectSetting.appName!
```

Comment:
Do we need to handle the case where appName is undefined?

Reply:
Per my understanding, most of the code will not be executed any more. So it's safe to treat this as step 1. After I merged this PR, will work with each feature owner to make necessary changes due to project settings clean up.